### PR TITLE
SUB-2106 Unique values

### PR DESCRIPTION
**Pull request by avrahams**
When V2ListRequest route option is set than add also 
POST <path>/uniqueValues 
with armotypes.UniqueValuesRequestV2 in body 
& UniqueValuesResponseV2 response 

Add uniqueValues tester
Add tests for notification cache api

### CODE REVIEW

File: db/utils.go [hunk 1]
```diff
@@ -154,6 +150,116 @@ func AdminFindPaginated[T any](c context.Context, findOps *FindOptions) (*types.
...
+		return nil, fmt.Errorf("findOps is nil")
+	}
+	if len(findOps.group) == 0 {
+		return nil, fmt.Errorf("group is empty")
+	}
+
+	//prepare a pipeline for each field
+	fieldsPipelines := map[string]mongoDB.Pipeline{}
+	for _, field := range findOps.group {
+		filedRef := "$" + field
```
Comment (Bezbran) [inline db/utils.go:179]: in goroutine

**Review by Bezbran — CHANGES_REQUESTED** [review]

**Review by Bezbran — CHANGES_REQUESTED** [review]

File: db/utils.go [hunk 2]
```diff
@@ -154,6 +150,110 @@ func AdminFindPaginated[T any](c context.Context, findOps *FindOptions) (*types.
...
+			results.Store(field, result)
+			return nil
+		})
+	}
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	//aggregate all fields results into one response
+	aggregatedResults := &armotypes.UniqueValuesResponseV2{
```
Comment (RinaO1234) [inline db/utils.go:225]: split to method

File: db/utils.go [hunk 3]
```diff
@@ -154,6 +150,110 @@ func AdminFindPaginated[T any](c context.Context, findOps *FindOptions) (*types.
...
+		Fields:      make(map[string][]string),
+		FieldsCount: make(map[string][]armotypes.UniqueValuesResponseFieldsCount),
+	}
+	var aggregateResultErr error
+	results.Range(func(key, value interface{}) bool {
+		field, ok := key.(string)
+		if !ok {
+			aggregateResultErr = errors.New("failed to cast key")
```
Comment (RinaO1234) [inline db/utils.go:233]: add more details to errors

File: db/utils.go [hunk 4]
```diff
@@ -154,6 +150,110 @@ func AdminFindPaginated[T any](c context.Context, findOps *FindOptions) (*types.
...
+	if findOps == nil {
+		findOps = &FindOptions{}
+	}
+	findOps.Filter().WithNotDeleteForCustomer(c)
+	return AdminAggregate[T](c, findOps)
+}
+
+// AdminAggregate search for docs of all customers (unless filtered by caller) and return aggregated result
+func AdminAggregate[T any](c context.Context, findOps *FindOptions) (*armotypes.UniqueValuesResponseV2, error) {
+	defer log.LogNTraceEnterExit(fmt.Sprintf("AdminAggregate %+v", findOps), c)()
```
Comment (RinaO1234) [inline db/utils.go:164]: why do we use defer ? and consider adding more debug logs before goroutines

**Review by RinaO1234 — APPROVED** [review]

File: db/utils.go (hunk 4 above)
Comment (RinaO1234) [inline db/utils.go:164]: cosider additional debug messages before / in gouruoutines, why do we use defer?

**Review by Bezbran — APPROVED** [review]

File: db/utils.go (hunk 1 above)
Comment (avrahams) [reply db/utils.go]: done

File: db/utils.go (hunk 4 above)
Comment (avrahams) [reply db/utils.go:164]: The defer is on the retuned function - so it prints a message (and add event to telementry) on enter the function and when existing it - for instance
```json
{"level":"info","ts":"2023-07-30T10:09:26.480561323+03:00","msg":"UpdateDocument","method":"PUT","query":"","path":"/v1_vulnerability_exception_policy","trace_id":"32e1c13ad87d8a273fd41c077f2fad25","span_id":"2ce5815fb7a55e20"}
{"level":"info","ts":"2023-07-30T10:09:26.481572213+03:00","msg":"UpdateDocument completed","method":"PUT","query":"","path":"/v1_vulnerability_exception_policy","trace_id":"32e1c13ad87d8a273fd41c077f2fad25","span_id":"2ce5815f
```

**Pull request merged**